### PR TITLE
User app override configs

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/giantswarm/issues/9037
These changes would utilize `user-override-apps` in the cluster namespace and would override the default app's properties with it for testing purpose. (e.g. testing charts in `default-test` catalog)

### CODE REVIEW

File: service/controller/key/key.go [hunk 1]
```diff
@@ -38,73 +37,6 @@ func CertConfigCertOperatorVersion(cr v1alpha1.CertConfig) string {
 	return cr.Labels[label.CertOperatorVersion]
 }
 
-// CommonAppSpecs returns apps installed for all providers.
-func CommonAppSpecs() []AppSpec {
```
Comment (tomahawk28) [inline service/controller/key/key.go]: Left over from last PRs.

File: service/controller/resource/app/desired.go [hunk 2]
```diff
@@ -94,6 +103,32 @@ func (r *Resource) getSecrets(ctx context.Context, cr apiv1alpha2.Cluster) (map[
...
+
+	appConfigs, ok := userConfig.Data[key.ReleaseVersion(&cr)]
+	if !ok {
+		// no release override configs, fall through
+		return nil, nil
+	}
+
+	err = yaml.Unmarshal([]byte(appConfigs), &u)
+	if err != nil {
+		r.logger.LogCtx(ctx, "level", "error", "message", "failed to unmarshal the user config", "stack", microerror.Stack(err))
```
Comment (tomahawk28) [inline service/controller/resource/app/desired.go]: If `configmap` is not following the format what we expected, just skipping.

**Comment by tomahawk28** [comment]
documentation in https://github.com/giantswarm/giantswarm/pull/9062

**Comment by tomahawk28** [comment]
Tested in `ginger`

File: service/controller/resource/app/desired.go [hunk 3]
```diff
@@ -182,6 +222,16 @@ func (r *Resource) newAppSpecs(ctx context.Context, cr apiv1alpha2.Cluster) ([]k
 			}
 		}
 
+		if val, ok := userOverrideConfigs[app.App]; ok {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found an user app Config for %#q, applying it", app.App))
```
Comment (rossf7) [inline service/controller/resource/app/desired.go]:
```suggestion
			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found a user override app config for %#q, applying it", app.App))
```

File: service/controller/resource/app/desired.go [hunk 4]
```diff
@@ -182,6 +222,16 @@ func (r *Resource) newAppSpecs(ctx context.Context, cr apiv1alpha2.Cluster) ([]k
 			}
 		}
 
+		if val, ok := userOverrideConfigs[app.App]; ok {
```
Comment (rossf7) [inline service/controller/resource/app/desired.go]: Can you add a comment so it's clear this is for overriding the config for manual testing? 

Similar to the comment above about overriding the app config.

**Comment by tomahawk28** [comment]
@rossf7 Docs are already merged, PTAL :)

**Pull request merged**